src/service.rs

```diff
@@ -594,21 +597,24 @@ impl Service {
             );
 
             // Check that the responder matches the expected request
-            if let Ok(request_node_address) = active_request.contact.node_address() {
-                if request_node_address != node_address {
-                    warn!("Received a response from an unexpected address. Expected {}, received {}, request_id {}", request_node_address, node_address, id);
-                    return;
-                }
+
+            let expected_node_address = active_request.contact.node_address();
+            if expected_node_address != node_address {
+                warn!("Received a response from an unexpected address. Expected {}, received {}, request_id {}", expected_node_address, node_address, id);
+                // TODO: here we removed the request. Check it
```

Comment:
Yes good point. The active_requests mapping heavily relies on the correctness of the handler. The handler also does this check, but in addition handles the timeouts for requests. 
I think in principle, if the handler works correctly, this is unreachable. But if we do reach this situation, we probably should fail loudly and we want to remove the mapping to avoid the mapping potentially growing indefinately

Reply:
added a loud debug fail and upgraded the log to err, not sure if you want to make it unreachable completely